core/main/src/service/apps/delegated_launcher_handler.rs

```diff
@@ -538,61 +534,80 @@ impl DelegatedLauncherHandler {
-            partner_id: String::from("partner.id.not.set"),
-            app_session_id: String::from("app_session_id.not.set"),
-            durable_app_id: app_id.to_string(),
-            app_version: None,
-            app_user_session_id: None,
-            governance_state: None,
-        };
-
-        /*
-        Do not forward internal errors from the launch handler as AppErrors. Only forward third-party application error messages as AppErrors.
-        TBD: Collaborate with the SIFT team to obtain the appropriate SIFT code for sharing error messages from the AppLauncher.
-        */
+            .endpoint_state
+            .has_rule("ripple.reportLifecycleStateChange")
+        {
+            let previous_state = self
+                .platform_state
+                .app_manager_state
+                .get_internal_state(app_id);
 
-        let inactive = self
-            .platform_state
-            .app_manager_state
-            .get(app_id)
-            .map_or(false, |app| app.initial_session.launch.inactive);
+            /*
+            Do not forward internal errors from the launch handler as AppErrors. Only forward third-party application error messages as AppErrors.
+            TBD: Collaborate with the SIFT team to obtain the appropriate SIFT code for sharing error messages from the AppLauncher.
+            */
 
-        let to_state = map_event(method, inactive);
+            let inactive = self
+                .platform_state
+                .app_manager_state
+                .get(app_id)
+                .map_or(false, |app| app.initial_session.launch.inactive);
 
-        if to_state.is_none() {
-            /*
-            NOOP, the launcher implementation does not care about this state w/respect to metrics */
-            return;
-        };
-        let from_state = match previous_state {
-            Some(state) => map_event(&state, inactive),
-            None => None,
-        };
+            let to_state = map_event(method, inactive);
 
-        let app_state_change_message =
-            BehavioralMetricPayload::AppStateChange(AppLifecycleStateChange {
-                context,
-                previous_state: from_state,
+            if to_state.is_none() {
+                /*
+                NOOP, the launcher implementation does not care about this state w/respect to metrics */
+                return;
+            };
+            let from_state = match previous_state {
+                Some(state) => map_event(&state, inactive),
+                None => None,
+            };
+            let params = serde_json::to_value(AppLifecycleStateChange {
+                app_id: app_id.to_owned(),
                 new_state: to_state.unwrap(),
-            });
+                previous_state: from_state,
+            })
+            .unwrap();
+            if BrokerUtils::process_for_app_main_request(
+                &mut self.platform_state.clone(),
+                "ripple.reportLifecycleStateChange",
```

Comment:
What's the reasoning for sending things like this through the rules engine instead of just directly reporting in Ripple, in other words what are we doing differently across, say, Sky and Comcast to warrant needing rules?

Reply:
We are trying to remove contract driven Metrics to use Json RPC driven metrics. This would help us realign extensions to use Json Rpc instead of FFI